hw/ip/rstmgr/rstmgr.core

```diff
@@ -10,8 +10,10 @@ filesets:
     depend:
       - lowrisc:ip:tlul
       - lowrisc:prim:clock_mux2
-      - lowrisc:ip:rstmgr_pkg
-      - fileset_top ? (lowrisc:systems:rstmgr)
+      - fileset_ip     ? (lowrisc:ip:rstmgr_pkg)
+      - fileset_top    ? (lowrisc:systems:rstmgr_pkg)
```

Comment:
hmmm.. .this seems a little weird to me... 
why is the rstmgr referencing the `systems` core file here?

Reply:
It's because the `top_earlgrey.core` does not depend on `systems:rstmgr` but only `ip:rstmgr`. Consequently, we have to select the rstmgr implementation/version inside `ip:rstmgr`.

I decided to clean this up a little. We were managing these kind of dependencies differently on an IP basis, which made it really hard to understand what's going on. Before the cleanup, we would check the flag both inside `rstmgr.core` and `rstmgr_pkg.core`. Now we just do it once inside `rstmgr.core`. The `systems:rstmgr_pkg` and `ip:rstmgr_pkg` then don't need to check the flags. This is more in line with what we do for other IPs.

Reply:
so actually i originally did it inside `rstmgr_pkg.core` intentionally.  Because there are other modules that reference rstmgr, and i didn't want them to have to decide whether they should pick the ip version or system, so instead i always pointed them at the ip one, and within the ip one used the flag to make a separation. 

It feels like we should at least add back the references in `rstmgr_pkg.core` because that might become a dependency for others? (maybe there's nothing at the moment...although i thought there was from pwrmgr).

Reply:
> Because there are other modules that reference rstmgr, and i didn't want them to have to decide whether they should pick the ip version or system, so instead i always pointed them at the ip one, and within the ip one used the flag to make a separation.

Glad to hear that. I think we are on the same page then.

Sorry Tim, I didn't know these dependencies were there on purpose - it was hard to figure out what's what. Before, we basically had
```
            /- systems:rstmgr_pkg
ip:rstmgr: -|
            \- ip:rstmgr_pkg
```
and for the package:
```
                /- systems:rstmgr_pkg (dependency)
ip:rstmgr_pkg: -|
                \- rtl/rstmgr_pkg.sv (ip source file)
```
And the same inside reset manager register core. There was no single dependency on `rstmgr_pkg` or `rstmgr_reg` only and thus I didn't see the reason why doing the same distinction in the package and register again would make sense.

Now everything is done inside `rstmgr_core`. There are no flags inside the package and register core files which simplifies things. The only dependencies I had to change were for IPs that are top-level specific/auto-generated anyway.

I believe I could change this back if preferred, but it might mess up again the dependency resolution for the topgen generator.

Reply:
For the record: We have decided to leave this as it is, i.e., to fan out inside `rstmgr`, `rstmgr_pgk` and `rstmgr_reg`. The main rationale behind this is that for DV, we don't want to pull in too much code but just the headers.